src/sections/wallet/assets/table/actions/WalletAssetsTableActions.tsx

```diff
@@ -16,168 +15,206 @@ import { useSetAsFeePayment } from "api/payments"
+          tOptions={{
+            asset: symbol,
+          }}
+        >
+          <span />
+          <span className="highlight" />
+        </Trans>
+      ),
+      onError: (
+        <Trans
+          t={t}
+          i18nKey="wallet.assets.table.actions.payment.toast.onLoading"
+          tOptions={{
+            asset: symbol,
+          }}
+        >
+          <span />
+          <span className="highlight" />
+        </Trans>
+      ),
+    })
+
+  const buttons: TDropdownItem[] = [
+    {
+      key: "trade",
+      icon: <TradeIcon />,
+      label: t("wallet.assets.table.actions.trade"),
+      onSelect: () =>
+        inTradeRouter
+          ? () =>
+              navigate({
+                to: "/trade/swap",
+                search: canBuy ? { assetOut: id } : { assetIn: id },
+              })
+          : undefined,
+      disabled: !inTradeRouter || account?.isExternalWalletConnected,
+    },
+    {
+      key: "transfer",
+      icon: <TransferIcon />,
+      label: t("wallet.assets.table.actions.transfer"),
+      onSelect: () => props.onTransferClick(),
+      disabled: account?.isExternalWalletConnected,
+    },
+    {
+      key: "deposit",
+      icon: <PlusIcon />,
+      label: t("wallet.assets.table.actions.deposit"),
+      onSelect: () => navigate({ to: LINKS.cross_chain }),
+      disabled: account?.isExternalWalletConnected,
+    },
+  ]
+
+  const hiddenBtnKeys = useMemo(() => {
+    return Object.entries(visible).reduce<string[]>(
+      (memo, [item, isVisible]) => {
+        if (!isVisible) memo.push(item)
+
+        return memo
+      },
+      [],
+    )
+  }, [visible])
 
   const actionItems = [
-    /*{
-      key: "add",
-      icon: <ClaimIcon />,
-      label: t("wallet.assets.table.actions.claim"),
-    },*/
-    !isLargeDesktop
-      ? {
-          key: "transfer",
-          icon: <TransferIcon />,
-          label: t("wallet.assets.table.actions.transfer"),
-        }
-      : null,
-    couldBeSetAsPaymentFee
+    enablePaymentFee
       ? {
           key: "setAsFeePayment",
           icon: <DollarIcon />,
           label: t("wallet.assets.table.actions.payment.asset"),
+          onSelect: () => onFeePaymentSelect(),
         }
       : null,
     couldWatchMetaMaskAsset
       ? {
           key: "watch",
           icon: <MetamaskLogo width={18} height={18} />,
           label: t("wallet.assets.table.actions.watch"),
+          onSelect: () =>
+            watchAsset(window?.ethereum, id, {
+              symbol: symbol,
+              decimals: decimals,
+            }),
         }
       : null,
   ].filter(isNotNil)
 
   return (
-    <>
+    <div
+      sx={{
+        flex: "row",
+        gap: 10,
+        justify: "end",
+      }}
+    >
       <div
         sx={{
           flex: "row",
           gap: 10,
-          display: ["none", "flex"],
-          align: "center",
+          flexWrap: "wrap",
+          height: 36,
```

Comment:
Should be 38px to align it with "more" button

![image](https://github.com/galacticcouncil/HydraDX-ui/assets/3890412/5f64053a-b93a-406d-b03f-9e6e103125f7)